src/librbd/ExclusiveLock.cc

```diff
@@ -295,10 +295,15 @@ void ExclusiveLock<I>::post_acquire_lock_handler(int r, Context *on_finish) {
 
   std::lock_guard locker{ML<I>::m_lock};
   m_pre_post_callback = on_finish;
+  bool discard_image_cache = false;
```

Comment:
Nit: I would like to avoid any hooks into other places when the only thing you need to do is delete the cache on-disk and clear some feature bits. It's probably just easier and cleaner to create a new state machine to handle this case.

Reply:
@dillaman Do you mean to create a new state in the state machine of ExclusiveLock?

Reply:
I think the flow should be from the API to a new RWL state machine to invalidate the cache. I don't see why you need to acquire the exclusive lock. Or if you do, it should be perhaps just a flag to state that you should not attempt to open the RWL cache when acquiring the lock.

Reply:
@dillaman  With exclusive lock to discard the RWL cache is to avoid discarding cache while the image is being used by other clients. If you think it is ok without an exclusive lock, I will skip the step to acquire the exclusive lock.     
Let me clarify the function to be added in rbd_invalidate_cache():  
1. If image_ctx->image_cache is not null, it discards the current data in RWL cache and closes the image cache.   
2. If image_ctx->image_cache is null (not opened) and the dirty_cache feature bit is set, just reset the feature bit and delete the cache file if it exists in this server.   
Do you think it is reasonable?

Reply:
Yes, sounds good to me.